### PR TITLE
[ts-bodies] First steps related to converter.ts compatibility

### PR DESCRIPTION
**Pull request by trilis**
* Add `TypescriptDukatCompilationTest` test set
* Fix if/else issues
* Add for-if statements
* Add initializers to properties

### CODE REVIEW

File: translator-string/src/org/jetbrains/dukat/translatorString/StringTranslator.kt [hunk 1]
```diff
@@ -341,16 +342,31 @@ private fun StatementModel.translate(): List<String> {
...
         is IfStatementModel -> {
-            val header = "if (${condition.translate()}) "
-            val mainBranch = thenStatement.translate()
-            val elseBranch = elseStatement?.translate() ?: listOf()
-            return listOf(header + mainBranch[0]) + mainBranch.drop(1) + elseBranch
+            val header = "if (${condition.translate()}) {"
+            val mainBranch = thenStatement.statements.flatMap { it.translate() }.map { FORMAT_TAB + it }
+            return when {
+                elseStatement == null -> listOf(header) + mainBranch + listOf("}")
+                elseStatement!!.statements.size == 1 && elseStatement!!.statements[0] is IfStatementModel -> {
```
Comment (shabunc) [inline translator-string/src/org/jetbrains/dukat/translatorString/StringTranslator.kt:349]: let's cast just once or at least reduce number of casts here
also, better to user firstOrNull

File: typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/introduceModels.kt [hunk 2]
```diff
@@ -351,6 +351,9 @@ internal class DocumentConverter(private val moduleNode: ModuleNode, private val
                     static = static,
                     override = null,
                     immutable = getter && !setter,
+                    initializer = with(ExpressionConverter(this@DocumentConverter)) {
```
Comment (shabunc) [inline typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/introduceModels.kt:354]: do we really need with here?

Comment (trilis) [reply typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/introduceModels.kt:354]: What's the alternative?

Comment (trilis) [reply typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/introduceModels.kt:354]: We need to call an extension function defined in other class, I don't know other way to do this

**Comment by trilis** [comment]
21ab49ac639bbf97a60e9eef2615384bff5725a3

**Pull request closed without merging**